.github/workflows/test.yml

```diff
@@ -43,9 +44,21 @@ jobs:
         ./scripts/poetry_test.bat
     - name: Upload coverage to Codecov
       if: matrix.os == 'ubuntu-latest'
-      uses: codecov/codecov-action@v1
+      uses: codecov/codecov-action@v4
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
         file: ./coverage.xml
         flags: unittests
-#        fail_ci_if_error: true
+  check: # This job does nothing and is only used for the branch protection
+    if: github.event.pull_request.draft == false
+    
+    needs:
+      - test
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Decide whether the needed jobs succeeded or failed
+        uses: re-actors/alls-green@release/v1 # nosemgrep
```

Comment:
This is the feature I wanted, and I would like to adopt it in datamodel-code-generator.

Reply:
Sure, please do. We use this also in our open source project FastStream - https://github.com/airtai/faststream/blob/main/.github/workflows/test.yaml#L476